client/lib/posts/actions.js

```diff
@@ -401,39 +401,43 @@ PostActions = {
 	/**
 	 * Calls out to API to update Post object with any changed attributes
 	 *
+	 * @param {Object} site Site object
 	 * @param {object} post to be changed
 	 * @param {object} attributes only send the attributes to be changed
 	 * @param {function} callback callback receives ( err, post ) arguments
+
 	 */
-	update: function( post, attributes, callback ) {
+	update: function( site, post, attributes, callback ) {
 		var postHandle = wpcom.site( post.site_ID ).post( post.ID );
 
-		postHandle.update( attributes, PostActions.receiveUpdate.bind( null, callback ) );
+		postHandle.update( attributes, PostActions.receiveUpdate.bind( site, null, callback ) );
```

Comment:
You went a little overboard with the `site` reposition. 😄 

On each of the three `receiveUpdate.bind`, `site` should be the second parameter, as the first is the `this` binding.
E.g. `receiveUpdate.bind( null, site, callback )`

Otherwise, we'd be passing `null` as the site.

Which, surprisingly, doesn't seem to break anything... 🤔
But, to be fair, my tests weren't that deep...

Reply:
Sorry, this mistake during the refactor, it followed the same logic for all in order to try not miss any case and I did not watch carefully this case.